README.md

```diff
@@ -242,3 +242,70 @@ const queryBuilder = this.repository
 
 return paginateRaw(queryBuilder, options);
 ```
+
+### Raw and Entities
+
+A similar approach is used for TypeORM's `getRawAndEntities`
+
+Let's assume there's a joined table that matches each cat with its cat toys.
+And we want to bring how many toys each cat has.
+
+```typescript
+
+const queryBuilder = this.repository
+  .createQueryBuilder<{ type: string; totalLives: string }>('cat')
+    .leftJoinAndSelect('cat.toys', 'toys')
+    .addSelect('COUNT(toys)::INTEGER', 'toyCount')
+    .groupBy('cat.name');
+```
+
+This will allow us to get the paginated cats information with the additional raw query to build our actual response value.
+The return pagination object will be the same, but you're now able to handle or map the results and the raw objects as needed.
+
+```typescript
+const [results, rawResults] = await paginateRawAndEntities(query, options);
+
+// we can do what we need with the items and raw results
+const newArray = doSomethingHere(results.items, rawResults),
+
+return new Pagination(
+  newArray,
+  results.meta,
+  results.links,
+);
```

Comment:
```suggestion
const [pagination, rawResults] = await paginateRawAndEntities(query, options);

pagination.items.map((item, index) => {
  // we can do what we need with the items and raw results here
  // change your items using rawResults[index] object
});

return pagination;
```

Reply:
I like this example, but I'm a bit worried people might access `rawResults[index]` and expect a perfect map, with the item (entity) they are handling which is not necessarily the case, especially with `join`ed queries. Just want to get your thoughts about that.

I can commit this change otherwise so you can merge the PR (I don't have write permission)

Reply:
sure I understand. Shall we put `rawResults.find(raw => raw.id === item.id)`?